src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3642,12 +3648,12 @@ IOperation popTarget(IOperation originalTarget, ImmutableArray<IArgumentOperatio
                     case OperationKind.PropertyReference:
                         var propertyReference = (IPropertyReferenceOperation)originalTarget;
                         Debug.Assert(propertyReference.Arguments.Length == arguments.Length);
-                        return new PropertyReferenceExpression(propertyReference.Property, instance, arguments, semanticModel: null, propertyReference.Syntax,
+                        var castArguments = arguments.CastArray<IArgumentOperation>();
```

Comment:
>CastArray [](start = 54, length = 9)

Do we also have a compiler verifiable down cast helper?

Reply:
Adding one. It'll just call CastArray under the hood, but it will verify that we're casting to a valid derived type.

---
In reply to: [186242445](https://github.com/dotnet/roslyn/pull/26635#discussion_r186242445) [](ancestors = 186242445)